packages/shared/src/UI/components/WalletStatusBar/WalletBarButton.tsx

```diff
@@ -0,0 +1,75 @@
+import { useCurrentWeb3NetworkPluginID } from '@masknet/plugin-infra/web3'
+import { makeStyles, MaskColorVar, useStylesExtends } from '@masknet/theme'
+import { Button, CircularProgress } from '@mui/material'
+import classNames from 'classnames'
+import { useState } from 'react'
+import { useSharedI18N } from '../../../locales'
+import type { WalletButtonActionProps } from './types'
+
+interface WalletButtonProps extends withClasses<'root'> {
+    actionProps?: WalletButtonActionProps
+    onConnectWallet?: () => void
+}
+
+const useStyles = makeStyles<{ color?: 'warning' }>()((theme, props) => ({
+    progress: {
+        color: MaskColorVar.twitterButtonText,
+        position: 'absolute',
+        top: theme.spacing(1),
+        left: `calc(50%-${theme.spacing(1)})`,
+    },
+    button: {
+        backgroundColor: props.color === 'warning' ? MaskColorVar.errorPlugin : '',
+        color: props.color === 'warning' ? '#FFFFFF' : '',
+        '&:hover': {
+            backgroundColor: props.color === 'warning' ? MaskColorVar.errorPlugin : '',
+        },
```

Comment:
Why do you need these judgments, if you need the warning style you can just add the color property to the button.